cg/services/file_delivery/file_formatter_service/delivery_file_formatter.py

```diff
@@ -0,0 +1,56 @@
+import os
+from pathlib import Path
+from cg.constants.delivery import INBOX_NAME
+from cg.services.file_delivery.fetch_file_service.models import DeliveryFiles
+from cg.services.file_delivery.file_formatter_service.utils.case_file_formatter import (
+    CaseFileFormatter,
+)
+from cg.services.file_delivery.file_formatter_service.delivery_file_formatting_service import (
+    DeliveryFileFormattingService,
+)
+from cg.services.file_delivery.file_formatter_service.models import FormattedFiles, FormattedFile
+from cg.services.file_delivery.file_formatter_service.utils.sample_file_concatenation_formatter import (
+    SampleFileConcatenationFormatter,
+)
+from cg.services.file_delivery.file_formatter_service.utils.sample_file_formatter import (
+    SampleFileFormatter,
+)
+from cg.services.file_delivery.file_formatter_service.utils.utils import (
+    get_ticket_dir_path,
+    create_ticket_dir,
+)
+
+
+class DeliveryFileFormatter(DeliveryFileFormattingService):
+    """
+    Format the files to be delivered in the generic format.
+    Expected structure:
+    <customer>/inbox/<ticket_id>/<case_name>/<case_files>
+    <customer>/inbox/<ticket_id>/<sample_name>/<sample_files>
+    """
+
+    def __init__(
+        self,
+        case_file_formatter: CaseFileFormatter,
+        sample_file_formatter: SampleFileFormatter | SampleFileConcatenationFormatter,
+    ):
+        self.case_file_formatter = case_file_formatter
+        self.sample_file_formatter = sample_file_formatter
+
+    def format_files(self, delivery_files: DeliveryFiles) -> FormattedFiles:
+        """Format the files to be delivered and return the formatted files in the generic format."""
+        ticket_dir_path: Path = get_ticket_dir_path(delivery_files.sample_files[0].file_path)
```

Comment:
This formatter gets a list of MOVED delivery files that are placed into cust/inbox/ticket. So the parent will always be that in this instance.

I can make this clearer to by perhaps renaming the parameter.

Reply:
But then the function `get_ticket_dir_path` makes assumptions on which type of path it will receive and not return the ticket_dir_path if we give it a slightly different path.  I would rather see a .parts on the path, find where the inbox is and reconstruct it. ChatGPT suggested something similar when i asked (see below).

```python
from pathlib import Path

def get_folder_under_inbox(path: Path) -> Path:
    # Convert to an absolute path, just in case
    path = path.resolve()
    
    # Find the "inbox" part of the path
    try:
        # Slice the path starting from the "inbox" directory
        inbox_index = path.parts.index('inbox')
        return Path(*path.parts[:inbox_index+2])
    except ValueError:
        raise ValueError("The specified path does not contain an 'inbox' directory")

# Example usage:
path = Path('/home/something/cust000/inbox/789789/case/sample')
result = get_folder_under_inbox(path)
print(result)

```

Reply:
What assumptions are made? Or you mean that it will not solve any problem with discerning if it should be `/home/something/cust000/inbox/789789/case/sample` or `/home/something/cust000/inbox/789789/sample`?